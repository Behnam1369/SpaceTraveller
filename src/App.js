import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Header from './components/Header';
import MyProfile from './components/MyProfile';
import './App.scss';
import MissionList from './components/MissionList';

function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Header />
        <Routes>
          <Route path="/MyProfile" element={<MyProfile />} />
          <Route path="/Missions" element={<MissionList />} />
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
