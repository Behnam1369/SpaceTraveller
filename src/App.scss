body {
  header {
    padding: 30px;

    .logo {
      width: 40px;
      height: 40px;
    }

    nav {
      float: right;
    }

    nav > a {
      text-decoration: none;
      color: blue;
      margin: 20px;
    }

    nav > a.active {
      text-decoration: underline;
      color: red;
    }
  }

  .missionList {
    width: 90%;
    margin: auto;
    border-collapse: collapse;

    thead {
      th {
        border: solid 1px silver;
        font-family: Helvetica, sans-serif;
        padding: 5px;
      }
    }

    tbody tr:nth-child(odd) {
      background-color: whitesmoke;
    }

    td {
      border: solid 1px silver;
      font-family: Helvetica, sans-serif;
      padding: 5px;

      span {
        background-color: gray;
        color: white;
        white-space: nowrap;
        padding: 5px;
        font-family: Helvetica, sans-serif;
        font-size: 14px;
        border-radius: 3px;
      }

      span.joined {
        background-color: darkcyan;
      }
    }

    td:nth-child(1) {
      font-weight: bold;
      text-align: center;
    }

    td:nth-child(2) {
      font-family: Arial, sans-serif;
      text-align: left;

      button {
        background-color: transparent;
        border: none;
        color: blue;
        cursor: pointer;
      }
    }

    td:nth-child(4) button {
      background-color: white;
      color: gray;
      border: solid 1px gray;
      white-space: nowrap;
      padding: 5px;
      font-family: Helvetica, sans-serif;
      font-size: 14px;
      cursor: pointer;
      border-radius: 4px;
    }

    td:nth-child(4) button.joined {
      color: red;
      border-color: red;
    }
  }

  .profile {
    margin: 7%;

    h1 {
      font-family: Helvetica, sans-serif;
    }

    h2 {
      font-family: Helvetica, sans-serif;
    }

    ul {
      width: 50%;
      margin: 0;
      padding: 0;

      li {
        list-style: none;
        margin: 0;
        padding: 20px;
        border: solid 1px lightgray;
      }
    }
  }
}
