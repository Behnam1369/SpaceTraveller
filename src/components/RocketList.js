import React from 'react';
import PropTypes from 'prop-types';
import Rocket from './Rocket';

const RocketList = (props) => {
  const { list } = props;
  return (
    <div className="rocket-list">
      <h1>Missions </h1>
      {list.map((el) => (
        <Rocket
          key={el.id}
          id={el.id}
          rocketName={el.rocket_name}
          description={el.description}
          image={el.flickr_images[0]}
        />
      ))}
    </div>
  );
};

RocketList.propTypes = {
  list: PropTypes.arrayOf(PropTypes.shape()).isRequired,
};

export default RocketList;
