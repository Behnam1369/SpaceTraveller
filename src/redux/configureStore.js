import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import missionReducer from './missions';
import { rocketsReducer } from './rockets';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionReducer,
});

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
