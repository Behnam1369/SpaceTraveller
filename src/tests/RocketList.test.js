import TestRenderer from 'react-test-renderer';
import {
  render,
  screen,
  fireEvent,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import RocketList from '../components/RocketList';
import '@testing-library/jest-dom';
import configureStore from '../redux/configureStore';
import App from '../App';

describe('Unit Testing Mission', () => {
  it('Test Mission component snapshot', () => {
    const testRenderer = TestRenderer.create(
      <Provider store={configureStore}>
        <RocketList />
      </Provider>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});

describe('Testing Rockets in Application', () => {
  it('Test reserve functionality', () => {
    const list = [
      {
        id: 'key1',
        name: 'TEST ROCKET',
        description: 'TEST DESCRIPTION',
        image: 'https://imgur.com/DaCfMsj.jpg',
        reserved: false,
      },
    ];

    configureStore.dispatch({ type: 'space-traveller/rockets/GET_API_DATA', data: list });

    render(
      <Provider store={configureStore}>
        <App />
      </Provider>,
    );

    fireEvent.click(screen.getByText('Reserve now!'));
    expect(screen.getByTestId('teal')).toHaveTextContent('reserved');
  });

  it('Test remove reservation functionality', () => {
    const list = [
      {
        id: 'key1',
        name: 'TEST ROCKET',
        description: 'TEST DESCRIPTION',
        image: 'https://imgur.com/DaCfMsj.jpg',
        reserved: true,
      },
    ];

    configureStore.dispatch({ type: 'space-traveller/rockets/GET_API_DATA', data: list });
    render(
      <Provider store={configureStore}>
        <App />
      </Provider>,
    );
    fireEvent.click(screen.getByText('Remove reservation'));
    expect(screen.getByTestId('quit')).toHaveTextContent('Reserve now!');
  });
});
