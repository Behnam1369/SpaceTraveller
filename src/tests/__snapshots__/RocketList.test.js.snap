// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing Mission Test Mission component snapshot 1`] = `
<div
  className="rocket-list"
/>
`;
